### DISCUSSION TITLE
How to build (Tabs, Carousel, Slider ... etc) cell-plugins? Or Custom Row/Cell Rendering?

### DISCUSSION THREAD

**Discussion by NedalEldeen** [react-page/react-page#1222] [General]
I've read some issues and PR discussions that talked, at some levels, about how we can build tabs, carousel, slider ... etc. I also read the answers that were expressing some restrictions to achieving such a thing at the moment. **But what makes me curious is;** in the main README.md file of this repo, [in the wild section](https://github.com/react-page/react-page#in-the-wild), the following point;

[Veloplus Online Shop](https://www.veloplus.ch/) - ReactPage is used to create and display any content page **(and the landing page as well)**

On the landing page of [Veloplus Online Shop](https://www.veloplus.ch/), there are some sliders there. **My question is;** Did they use `react-page` to make those sliders? If so, how did they achieve that? Does anyone know anything about it?

I know that the `react-page` editor is great and it has a lot of potential to be a powerful page editor. And I think it would be nice and more powerful if;

- It's easy to build such a thing.
- Custom Row/Cell Rendering.
- Rich documentation with more examples.
- Well-defined roadmap for this project.
- ... .

I'm sure the `react-page` team has a vision for this editor and has a lot to add. And I started this discussion because I hope that we can talk more about these points.

**Comment by macrozone** [comment]
@NedalEldeen (disclaimer: i am lead developer for the frontend of https://www.veloplus.ch):

the slider on the landing page (for example those that list products) are react-components, something like `<ProductSlider productIds={productIds} />`,

You can then [create a CellPlugin](https://react-page.github.io/docs/#/custom-cell-plugins) to make this slider available for your editor-users. You will probably want the form of this plugin to render a list of inputs, so that users can add productIds that will show the product in the slider.

A simplifed example of this is on the repository and shown on the demo-page here: https://react-page.github.io/examples/reactadmin#/posts/post1 ("our recommended products")


> Rich documentation with more examples.

Yeah, the documentation can still be improved with better examples. I am happy to accept Pull request that help improving it!
Everyone is also welcomed to show usecases here in the discussion board or have good Q&As about specific questions

**Comment by NedalEldeen** [comment]
@macrozone, thanks for your answer. I thought **veloplus** using a different approach rather than rendering an array of items. Let me put it in a different way. Assume we need to make a slider and each slide has its own inner components. Look at the following diagram;

<img width="1626" alt="Screen Shot 2022-08-17 at 11 59 48 PM" src="https://user-images.githubusercontent.com/26748456/185251876-ad1a4d5f-e5c6-48ef-a530-f6fc4e98d8a3.png">


What if I have a `sliderCell` and inside this cell, I can add `sliderItemCell`. And each `sliderItemCell` can add any other allowed cells like text, image, grid, video ... etc.


**Another use case;** Implementing tabs.

<img width="1509" alt="Screen Shot 2022-08-17 at 11 59 28 PM" src="https://user-images.githubusercontent.com/26748456/185252613-e48d3701-9cd6-4b77-b5d5-2fa196153cf3.png">




**Is it possible to achieve something like that? Even if in the future?**

**Comment by NedalEldeen** [comment]
For following up, please follow this issue https://github.com/react-page/react-page/issues/1231
